js.go

```diff
@@ -1442,10 +1463,41 @@ func (js *js) subscribe(subj, queue string, cb MsgHandler, ch chan *Msg, isSync,
 	if hasHeartbeats {
 		sub.scheduleHeartbeatCheck()
 	}
+	// For ChanSubscriptions, if we know that there is flow control, we will
+	// start a go routine that evaluates the number of delivered messages
+	// and process flow control.
+	if sub.Type() == ChanSubscription && hasFC {
+		go sub.chanSubcheckForFlowControlResponse()
+	}
 
 	return sub, nil
 }
 
+// This long-lived routine is used per ChanSubscription to check
+// on the number of delivered messages and check for flow control response.
+func (sub *Subscription) chanSubcheckForFlowControlResponse() {
+	sub.mu.Lock()
+	nc := sub.conn
+	sub.mu.Unlock()
+	if nc == nil {
+		return
+	}
+	t := time.NewTicker(chanSubFCCheckInterval)
```

Comment:
Should probably have a chan here that is closed when sub is unsubscribed or connection is closed.

Reply:
Could add a chan too (in addition to the timer). I thought that it was ok like this, meaning that this go routine may linger for 250ms and then we would detect that the subscription is gone.

Reply:
Since I use timer now, it is stopped when the subscription is stopped/removed.